src/ints/int10_modes.cpp

```diff
@@ -2162,23 +2160,23 @@ std::optional<Rgb666> parse_color_token(const std::string& token,
 				log_warning("could not parse RGB-triplet value");
 				return -1;
 			}
-			if (value > 0xff) {
+			if (value < 0 || value > 255) {
 				log_warning("RGB-triplet values must be between 0 and 255");
 				return -1;
 			}
 			return value;
 		};
 
-		const auto r = parse_component(r_string);
-		const auto g = parse_component(g_string);
-		const auto b = parse_component(b_string);
+		const auto r8 = parse_component(r_string);
+		const auto g8 = parse_component(g_string);
+		const auto b8 = parse_component(b_string);
 
-		if (r < 0 || g < 0 || b < 0) {
+		if (r8 < 0 || g8 < 0 || b8 < 0) {
```

Comment:
I guess this is user-data coming (custom CGA palette stuff), so we're not certain of these integer values. Suggest adding upper-bound checks to make sure `values are < 256` so they're not wrapped in the convert to Rgb888.

Reply:
It' handled in the parser lambda a few lines above. This is the error path because the one or all the components are set to negative to signal an error. But I'll change it to optional 😄